src/server/bots/bridgeless/txidSubmissionService.ts

```diff
@@ -0,0 +1,203 @@
+import { asNumber, asObject, asOptional, asValue } from 'cleaners'
+
+import type { BridgelessSubmission } from './types'
+
+const doFetch = async (
+  url: string,
+  opts: RequestInit = {}
+): Promise<unknown> => {
+  const res = await fetch(url, opts)
+  if (!res.ok) {
+    const message = await res.text()
+    throw new Error(`Fetch failed: ${message}`)
+  }
+  const json = await res.json()
+  return json
+}
+
+const asBridgelessNumConfirmations = asObject({
+  chain: asObject({
+    // "id": "0",
+    // "type": "BITCOIN",
+    // "bridge_address": "1E3TeJbW5iy6b2YLF427Za4HTaYQ3yFSUK",
+    // "operator": "1E3TeJbW5iy6b2YLF427Za4HTaYQ3yFSUK",
+    confirmations: asNumber
+    // "name": "Bitcoin"
+  })
+})
+const getRequiredConfirmations = async (chainId: string): Promise<number> => {
+  const json = await doFetch(
+    `https://rpc-api.node0.mainnet.bridgeless.com/cosmos/bridge/chains/${chainId}`
+  )
+  const clean = asBridgelessNumConfirmations(json)
+  return clean.chain.confirmations
+}
+
+const BITCOIN_BLOCKBOOK_URL = 'https://btc-wusa1.edge.app/api/v2'
+
+const asBitcoinBlockbookInfo = asObject({
+  blockbook: asObject({
+    bestHeight: asNumber
+  })
+})
+const getBitcoinChainHeight = async (): Promise<number> => {
+  const json = await doFetch(`${BITCOIN_BLOCKBOOK_URL}/`)
+  const clean = asBitcoinBlockbookInfo(json)
+  return clean.blockbook.bestHeight
+}
+
+const asBitcoinTransaction = asObject({
+  blockHeight: asOptional(asNumber),
+  confirmations: asOptional(asNumber)
+})
+const getBitcoinTransactionHeight = async (txid: string): Promise<number> => {
+  const json = await doFetch(`${BITCOIN_BLOCKBOOK_URL}/tx/${txid}`)
+  const clean = asBitcoinTransaction(json)
+  if (
+    clean.blockHeight != null &&
+    clean.confirmations != null &&
```

Comment:
why even look at 'confirmations' you only use the blockHeight and it obviously needs to be > 0 before you care.

Reply:
I blame the AI that converted from mempool to blockbook